docs/grammar/Solidity.g4

```diff
@@ -189,20 +190,40 @@ locals[
 	(Semicolon | body=block);
 
 /**
- * Definitions of the special fallback and receive functions.
+ * Definitions of the special fallback function.
  */
-fallbackReceiveFunctionDefinition
+fallbackFunctionDefinition
 locals[
-	boolean visibilitySet = false,
 	boolean mutabilitySet = false,
+	boolean virtualSet = false,
+	boolean overrideSpecifierSet = false,
+	boolean hasParameters = false
+]
+:
+	kind=(Fallback | Receive) LParen (parameterList { $hasParameters = true; } )? RParen
+	(
+		External
```

Comment:
Doesn't this need `visibilitySet` guard to ensure it cannot appear multiple times?

Same in case of `External` and `Payable` in `receiveFunctionDefinition` below.

Reply:
Changed.

Reply:
I think you also need to restore the `boolean visibilitySet = false` line (and similarly `visibilitySet` + `mutabilitySet` in `receiveFunctionDefinition`).